crates/db_schema/src/impls/post.rs

```diff
@@ -341,6 +366,92 @@ impl Post {
     }
     Ok(())
   }
+
+  pub async fn fill_aggregates_history(pool: &mut DbPool<'_>) -> LemmyResult<()> {
+    let conn = &mut get_conn(pool).await?;
+
+    info!("Filling post_aggregates history into post...");
+
+    // Get the total count of post_aggregates rows, to show progress
+    let post_aggregates_count = post_aggregates::table
+      .select(count_star())
+      .first::<i64>(conn)
+      .await?;
+
+    let mut processed_rows = 0;
+
+    while processed_rows < post_aggregates_count {
+      let rows_updated = conn
+        .run_transaction(|conn| {
+          async move {
+            // Diesel can't do 'update X from Y', nor updates from joins, so you need to do custom
+            // sql. I also tried individual row sets, and it was too slow.
+            let updated_rows = sql_query(
+              r#"
+              WITH pa AS (SELECT *
+                FROM post_aggregates pa
+                ORDER BY pa.post_id desc
+                LIMIT $1)
+              UPDATE post p
+                SET
+                  comments = pa.comments,
+                  score = pa.score,
+                  upvotes = pa.upvotes,
+                  downvotes = pa.downvotes,
+                  newest_comment_time_necro_at = pa.newest_comment_time_necro,
+                  newest_comment_time_at = pa.newest_comment_time,
+                  hot_rank = pa.hot_rank,
+                  hot_rank_active = pa.hot_rank_active,
+                  controversy_rank = pa.controversy_rank,
+                  scaled_rank = pa.scaled_rank,
+                  report_count = pa.report_count,
+                  unresolved_report_count = pa.unresolved_report_count
+                FROM pa WHERE p.id = pa.post_id
+                RETURNING p.id;
+            "#,
+            )
+            .bind::<BigInt, _>(DB_BATCH_SIZE)
+            .get_results::<AggregatesUpdateResult>(conn)
+            .await?;
+
+            // When this is None, the scanning is complete
+            let last_scanned_id = updated_rows.last().map(|f| f.id);
```

Comment:
You are deleting the copied rows from post_aggregates anyway, so storing the id is unnecessary.